addons/stock_account/models/product.py

```diff
@@ -114,11 +109,6 @@ class ProductProduct(models.Model):
     stock_value = fields.Float(
         'Value', compute='_compute_stock_value')
 
-    @api.onchange('type')
-    def onchange_type_valuation(self):
-        # TO REMOVE IN MASTER
-        pass
-
     @api.multi
     def do_change_standard_price(self, new_price, account_id):
         """ Changes the Standard Price of Product and creates an account move accordingly."""
```

Comment:
location.valuation_in_account_id and location.valuation_out_account_id are only used when products are moved to/from production or inventory locations , isn't it. When adjusting valuation of products by using this method, accounts are coming from template/category accounts.

So for locations with type `internal` (or `transit` following your same approach of thinking transit is like internal) no accounts from there are used. 
At this point only locations with those two types are to be checked for qty_available,

Thus grouping is ok and it won't affect those locations using accounts.